### PR TITLE
Reduce (or increase) NPP due to LUC effects on vegetation

### PR DESCRIPTION
**Pull request by bpbond**
This PR cherry-picks the NPP reduction (due to LUC effects on vegetation) code from #639 .

This PR adds a new term to our NPP calculation: vegetation losses (or gains) due to LUC change, with the the aim of avoiding double-counting of regrowth fluxes. We save the end-of-spinup vegetation C value; keep a running total of LUC emissions from (or gains to) vegetation; and then adjust NPP:
```c++
  npp_luc_adjust = (end_of_spinup_vegc - cum_luc_va) / end_of_spinup_vegc;
  npp = npp * npp_luc_adjust;
```

Cumulative vegetation losses from LUC:

cum_luc_va

This peaks at ~80 Pg C, which ~15% of the end-of-spinup vegetation C value of 550 Pg C, and so the `npp_luc_adjust` drops to ~85%:

npp_luc_adjust

### CODE REVIEW

**Comment by bpbond** [comment]
Differences from `v3_dev`:

diff

Notice that NPP drops by, again, ~15%:

diff_percent

**Comment by bpbond** [comment]
One other thought: we could give users control over the strength of this effect. For example, in the INI files there could be
```
 luc_npp_effect=1.0
```
meaning that every petagram of vegetation loss from LUC results in a petagram of NPP reduction. Setting this parameter to 0.5 would mean a 0.5 Pg NPP reduction for every Pg veg loss, and setting to zero would replicate the behavior of Hector 2.x: NPP is unaffected by LUC.

**Comment by bpbond** [comment]
My thought was that `luc_npp_effect` wouldn't affect LUC emissions _at all_; it would only affect how much NPP is downregulated due to vegetation losses to LUC. Modifying the equations above:
```c++
npp_luc_adjust = (end_of_spinup_vegc - (cum_luc_va * luc_npp_effect)) / end_of_spinup_vegc;
npp = npp * npp_luc_adjust;
```

...so that when `luc_npp_effect` is 1.0, it's the same as the equation above; at 0.5, NPP would adjust downward at 50% the rate of veg loss; and when it's 0.0, `npp_luc_adjust` would be 1, and so `npp` wouldn't change at all.

**Comment by kdorheim** [comment]
> luc_npp_effect wouldn't affect LUC emissions at all;

Correct. Isn't this development about making sure NPP is adjusted with respect to the luc emissions? so like when `luc_npp_effect` is 0 luc would have no effect on NPP aka seeing the same behavior we are in Hector v2.5 that double counts carbon?

**Comment by bpbond** [comment]
Right; and right, if it was zero, you'd potentially be in a double-counting situation. (But maybe you're running an experiment where you _want_ NPP rebound from LUC.)

**Comment by bpbond** [comment]
This would also be super simple to add later; we could say the `npp_luc_effect` idea is something for 3.1 or 3.5.

**Review by kdorheim — APPROVED** [review]
I think this looks good, I like the idea of adding the uncertainty parameter post-v3, I think we will want to be mindful about setting up constraints on it/documentation to ensure that it is being used appropriately

**Comment by bpbond** [comment]
Sounds good, thanks.

File: tests/testthat/test_old-new.R [hunk 1]
```diff
@@ -6,7 +6,7 @@ context("Old new tests")
 # need to be updated by running data-raw/generate-comp-data.R
 
 test_that("Hector output passes old new test", {
-
+testthat::skip("skipping")
```
[the remark this answers is not in the document]
Comment (bpbond) [reply tests/testthat/test_old-new.R:9]: Yes, sorry, will do that.

**Comment by bpbond** [comment]
I assume you will check and merge when ready

**Comment by kdorheim** [comment]
Are we ready to merge?

**Comment by bpbond** [comment]
From my point of view, yes, but depending on you to assess model performance and decide you (we) are OK with it. Thanks!

**Comment by kdorheim** [comment]
@bpbond here are some comparisons between this latest Hector v3 with observations 

image
image
image
image

Future CO2 emissions vs gmst between IPCC and Hector is spot on! We are on the cooler end of the envelope during the historical period... which is not great but I am not sure how concerning that is...  
image

**Comment by bpbond** [comment]
@kdorheim Wow, that looks really good overall. The only thing concerning I see is the land sink graph — any insight into why we drop so abruptly around 2000?

Re temperature, how can we be so down the middle vs. NASA GISS but so low in the IPCC SPM 10 graph?

**Comment by kdorheim** [comment]
Those are great questions but unfortunately, I don't have great answers... 

re the land sink 
* could be caused by different luc emissions used over the historical period however I did try doing the calibration protocol with the GCB luc emissions and we still see that drop in 2020 
* we do see a large decrease in luc emissions starting in 2000 and NBP remains fairly stable GCB luc emissions are slightly higher during this period but even when I run with the GCB luc emissions driven runs we see the same behavior of underestimating the land sink starting around 2000
image

Screen Shot 2022-11-07 at 9 54 57 AM


In fact when we use GCB luc emissions as inputs the land sink fit is worse after the year 2000 and the [co2] also suffers


Re the IPCC figure 
*[Here](https://www.ipcc.ch/report/ar6/wg1/figures/summary-for-policymakers/figure-spm-10) is the information that I used to construct the figure. I normalized hector's temperature using Hector's average temperature from 1850-1900. Looking at the IPCC historical results, it does appear that their results were normalized to this period, the IPCC 1850 to 1900 is non 0 which to me suggests that we aren't using the same reference period/something is wrong with how we are normalizing.

**Comment by bpbond** [comment]
Thanks for the additional detail @kdorheim . Regardless, overall things look good to me.

**Comment by kdorheim** [comment]
After this passes tests I this this is ready to merge from my pov

**Comment by bpbond** [comment]
And then maybe we make a `v3.0.alpha` tag? 🎉

**Comment by kdorheim** [comment]
I don't think we are ready for the v3 alpha branch. I need to update the ini files with the new default parameters! But let's get this NPP branch merged into v3!

**Pull request merged**